### PR TITLE
Fixes #34377 - Use new process abstraction from sp-dynflow

### PR DESCRIPTION
**Pull request by adamruzicka**
Requires:
- https://github.com/theforeman/smart_proxy_dynflow/pull/103

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
[test smart_proxy_ansible]

**Comment by adamruzicka** [comment]
> Workflow runs completed with no jobs

Interesting

File: lib/smart_proxy_ansible/runner/ansible_runner.rb [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 require 'smart_proxy_dynflow/runner/base'
 require 'smart_proxy_dynflow/runner/parent'
 module Proxy::Ansible
   module Runner
     class AnsibleRunner < ::Proxy::Dynflow::Runner::Parent
-      include ::Proxy::Dynflow::Runner::Command
+      include ::Proxy::Dynflow::Runner::ProcessManagerCommand
       attr_reader :execution_timeout_interval, :command_pid
 
       def initialize(input, suspended_action:)
```
Comment (ezr-ondrej) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:13]: I know it is not strictly related, but now it is untestable with dynflow master, can we add the id: from [`On demand`](https://github.com/theforeman/smart_proxy_dynflow/pull/105) first so we won't forget (probably separate PR)?
```suggestion
      def initialize(input, suspended_action:, id: nil)
```

Comment (ezr-ondrej) [reply lib/smart_proxy_ansible/runner/ansible_runner.rb:13]: right... if I'd just read first :) it alrady implemented as #63

**Comment by ezr-ondrej** [comment]
egh, we have a conflict now :(

**Comment by adamruzicka** [comment]
Not anymore

**Comment by ezr-ondrej** [comment]
We are not initializing the ProcessManager, because we overide the `initialize_command` and thus refresh fails on `@process_manager.process` no method process for nil. should we call super in `initialize_command`?

**Comment by adamruzicka** [comment]
Right, I forgot we did https://github.com/theforeman/smart_proxy_ansible/pull/61 in the mean time

File: lib/smart_proxy_ansible/runner/ansible_runner.rb [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-require 'smart_proxy_dynflow/runner/command'
+require 'smart_proxy_dynflow/runner/process_manager_command'
 require 'smart_proxy_dynflow/runner/base'
 require 'smart_proxy_dynflow/runner/parent'
 module Proxy::Ansible
   module Runner
     class AnsibleRunner < ::Proxy::Dynflow::Runner::Parent
-      include ::Proxy::Dynflow::Runner::Command
+      include ::Proxy::Dynflow::Runner::ProcessManagerCommand
       attr_reader :execution_timeout_interval, :command_pid
```
Comment (ezr-ondrej) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:11]: Probably not needed now, right?
```suggestion
      attr_reader :execution_timeout_interval
```

File: lib/smart_proxy_ansible/runner/ansible_runner.rb [hunk 3]
```diff
@@ -77,6 +67,11 @@ def publish_exit_status(status)
         super
       end
 
+      def set_process_manager_callbacks(pm)
+        super
+        pm.stdin.close
```
Comment (ezr-ondrej) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:72]: This fails for me: https://pastebin.com/sBVC5cCc

**Comment by adamruzicka** [comment]
In case `ansible-runner` command is missing we will still run into https://projects.theforeman.org/issues/31888 , I'll address that in a followup PR

**Review by ezr-ondrej — APPROVED** [review]

**Pull request merged**